spec/std/path_spec.cr

```diff
@@ -0,0 +1,594 @@
+    block.call(Path.windows(path)).should eq(windows)
+  end
+end
+
+describe Path do
+  describe ".new" do
+    it { Path.new("foo").native?.should be_true }
+    it { Path.new("foo").to_s.should eq "foo" }
+
+    it "fails with null byte" do
+      expect_raises ArgumentError, "String contains null byte" do
+        Path.new("foo\0")
+      end
+    end
+
+    it { Path.new.to_s.should eq "" }
+  end
+
+  describe ".posix" do
+    it { Path.posix("foo").posix?.should be_true }
+    it { Path.posix("foo").windows?.should be_false }
+    it { Path.posix("foo").to_s.should eq "foo" }
+
+    it "fails with null byte" do
+      expect_raises ArgumentError, "String contains null byte" do
+        Path.posix("foo\0")
+      end
+    end
+
+    it { Path.posix.to_s.should eq "" }
+  end
+
+  describe ".windows" do
+    it { Path.windows("foo").posix?.should be_false }
+    it { Path.windows("foo").windows?.should be_true }
+    it { Path.windows("foo").to_s.should eq "foo" }
+
+    it "fails with null byte" do
+      expect_raises ArgumentError, "String contains null byte" do
+        Path.windows("foo\0")
+      end
+    end
+
+    it { Path.windows.to_s.should eq "" }
+  end
+
+  describe "{}" do
+    Path{"foo"}.should eq Path.new("foo")
+    Path{"foo"}.native?.should be_true
+  end
+
+  describe "literal" do
+    it { Path{"foo", "bar", "baz"}.should eq Path.new("foo", "bar", "baz") }
+    it { Path{"/foo", "bar", "baz"}.should eq Path.new("/foo", "bar", "baz") }
+  end
+
+  describe "#parent" do
+    assert_paths("/Users/foo/bar.cr", "/Users/foo", &.parent)
+    assert_paths("Users/foo/bar.cr", "Users/foo", &.parent)
+    assert_paths("foo/bar/", "foo", &.parent)
+    assert_paths("foo/bar/.", "foo/bar", &.parent)
+    assert_paths("foo/bar/..", "foo/bar", &.parent)
+    assert_paths("foo", ".", &.parent)
+    assert_paths("foo/", ".", &.parent)
+    assert_paths("/", "/", &.parent)
+    assert_paths("////", "/", &.parent)
+    assert_paths("foo//.//", "foo", &.parent)
+    assert_paths("/.", "/", &.parent)
+    assert_paths("/foo", "/", &.parent)
+    assert_paths("", ".", &.parent)
+    assert_paths("./foo", ".", &.parent)
+    assert_paths(".", ".", &.parent)
+    assert_paths("\\Users\\foo\\bar.cr", ".", "\\Users\\foo", &.parent)
+    assert_paths("\\Users/foo\\bar.cr", "\\Users", "\\Users/foo", &.parent)
+    assert_paths("foo\\bar\\", ".", "foo", &.parent)
+    assert_paths("foo\\bar\\.", ".", "foo\\bar", &.parent)
+    assert_paths("foo\\bar\\..", ".", "foo\\bar", &.parent)
+    assert_paths("foo\\", ".", &.parent)
+    assert_paths("\\", ".", "\\", &.parent)
+    assert_paths(".\\foo", ".", &.parent)
+    assert_paths("C:", ".", "C:", &.parent)
+    assert_paths("C:/", ".", "C:/", &.parent)
+    assert_paths("C:\\", ".", "C:\\", &.parent)
+    assert_paths("C:/foo", "C:", "C:/", &.parent)
+    assert_paths("C:\\foo", ".", "C:\\", &.parent)
+    assert_paths("/foo/C:/bar", "/foo/C:", "/foo/C:", &.parent)
+  end
+
+  describe "#parents" do
```

Comment:
there is no spec for accessing parents of paths that have `..` like `./foo/../../bar`. What should be its behavior? normalize first? or return components as parent?

Reply:
It should normalise

Reply:
I looked at python pathlib and it doesn't normalize the path either.

Reply:
I added a few specs for this